### PR TITLE
Refactor `kompile` and `KompiledTest`

### PR DESCRIPTION
**Pull request by tothtamas28**

### CODE REVIEW

File: src/integration_tests/kompiled_test.py [hunk 1]
```diff
@@ -2,49 +2,57 @@
...
+    KOMPILE_BACKEND: ClassVar[Optional[str]] = None
+    KOMPILE_MAIN_MODULE: ClassVar[Optional[str]] = None
+    KOMPILE_SYNTAX_MODULE: ClassVar[Optional[str]] = None
+    KOMPILE_INCLUDE_DIRS: ClassVar[Iterable[str]] = []
+    KOMPILE_POST_PROCESS: ClassVar[Optional[str]] = None
 
-    kompiled_dir: Path
-    definition: KDefinition
+    kompiled_dir: ClassVar[Path]
+    definition: ClassVar[KDefinition]
```
Comment (tothtamas28) [inline src/integration_tests/kompiled_test.py:24]: This ensures that kompilation occurs once per `KompiledTest` subclass, which results in a noticeable speedup for integration tests (around 30%).

Hopefully it won't cause issues for test parallelization. If it does, we can easily revert this change.

**Review by ehildenb — APPROVED** [review]

File: src/integration_tests/kompiled_test.py (hunk 1 above)
Comment (ehildenb) [reply src/integration_tests/kompiled_test.py:24]: Looks like it only affects the tests, rather than the main `kompile` method.

Comment (tothtamas28) [reply src/integration_tests/kompiled_test.py:24]: Yes, the change here is that now `kompile` runs before the first test only, and not before each test.
